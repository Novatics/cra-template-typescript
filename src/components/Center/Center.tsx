import React, { ReactNode } from 'react';

interface CenterProps {
  children: ReactNode;
}

const Center = ({ children }: CenterProps) => {
  return <div>{children}</div>;
};

export default Center;
