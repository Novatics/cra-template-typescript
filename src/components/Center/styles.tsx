import styled from 'styled-components'

export const StyledContainer = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`
