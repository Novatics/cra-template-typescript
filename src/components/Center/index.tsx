import React, { ReactNode } from 'react';
import { StyledContainer } from './styles';

interface CenterProps {
  children: ReactNode;
}

const Center: React.FC<CenterProps> = ({ children }: CenterProps) => {
  return <StyledContainer>{children}</StyledContainer>;
};

export default Center;
