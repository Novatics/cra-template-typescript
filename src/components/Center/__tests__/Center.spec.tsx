import React from 'react';
import { render } from '@testing-library/react';
import Center from '..';

describe('Center', () => {
  it('should take a snapshot', () => {
    const Render = () => <Center>Text</Center>;
    const { asFragment } = render(<Render />);

    expect(asFragment()).toMatchSnapshot();
  });
});
