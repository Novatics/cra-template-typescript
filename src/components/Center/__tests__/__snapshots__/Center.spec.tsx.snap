// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Center should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    Text
  </div>
</DocumentFragment>
`;
